### PR TITLE
aspino, avy, glucose: fix build

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

As in the title.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tjni** [comment]
Thank you very much for working on fixing this package.

Is it possible to create a PR in [upstream](https://github.com/alviano/aspino) and pull this in using `fetchpatch`? There is a chance it will be merged and, even if not, could serve to add more information about why this fixes the build.

Separately, would it be possible to check of both `badPlatforms` and `broken` attribute can be removed (which can be tested via `ofborg`)?

File: pkgs/applications/science/logic/aspino/default.nix [hunk 1]
```diff
@@ -23,7 +23,10 @@ stdenv.mkDerivation {
   postPatch = ''
     substituteInPlace Makefile \
       --replace "GCC = g++" "GCC = c++"
-
+    substituteInPlace src/main.cc \
+      --replace "defined(__linux__)" "defined(__linux__) && defined(__x86_64__)"
+    substituteInPlace src/MaxSatSolver.cc \
+      --replace "occ[i][sign(softLiterals[j])] > 0" "occ[i][sign(softLiterals[j])] != 0"
     patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/aspino/default.nix:30]: While we are at it: can we change this to a directory that needs patching?

Comment (wegank) [reply pkgs/applications/science/logic/aspino/default.nix:30]: I didn't get this: you mean move it to a patch file?

Comment (SuperSandro2000) [reply pkgs/applications/science/logic/aspino/default.nix:30]: No, change dot to a few files or directories instead of everything

Comment (wegank) [reply pkgs/applications/science/logic/aspino/default.nix]: I think we can even remove patchShebangs: #!/bin/sh seems perfectly legal to me.

**Pull request merged**